src/Dfe.PlanTech.Application/Content/Queries/GetRecommendationChunksQuery.cs

```diff
@@ -0,0 +1,25 @@
+using Dfe.PlanTech.Domain.Content.Queries;
+using Dfe.PlanTech.Domain.Questionnaire.Models;
+
+namespace Dfe.PlanTech.Application.Content.Queries;
+
+public class GetRecommendationChunksQuery(IGetSubTopicRecommendation getSubTopicRecommendationFromContentfulQuery) : IGetRecommendationChunks
+{
+    private readonly IGetSubTopicRecommendation _getSubTopicRecommendationFromContentfulQuery = getSubTopicRecommendationFromContentfulQuery;
+
+    public async Task<IEnumerable<RecommendationChunk>> GetRecommendationChunksFromAnswers(Section subTopic, IEnumerable<Answer> answers, CancellationToken cancellationToken = default)
+    {
+        SubTopicRecommendation subTopicRecommendation = await _getSubTopicRecommendationFromContentfulQuery.GetSubTopicRecommendation(subTopic, cancellationToken);
+
+        RecommendationSection recommendationSection = subTopicRecommendation.Section;
+
+        IEnumerable<RecommendationChunk> recommendationChunks = recommendationSection.Chunks;
+
+        if (recommendationSection.Answers.IntersectBy(answers.Select(answer => answer.Sys.Id), recAnswer => recAnswer.Sys.Id).Any())
+        {
+            return recommendationChunks;
+        }
+
+        return recommendationChunks.Where(chunk => chunk.Answers.IntersectBy(answers.Select(answer => answer.Sys.Id), chunkAns => chunkAns.Sys.Id).Any());
```

Comment:
IMO I would try refactor this and line 18 a bit to make it a bit easier to read.

Something like

1. `answers.Select(answer => answer.Sys.Id)` I'd make this a variable before line 18, as a a HashSet. So you've got `var answerIds = answers.Select(answer => answer.Sys.Id).ToHashSet()` which IMO would improve readability and _likely_ performance.
2. Refactor both lines where they say:

```csharp
Answers.IntersectBy(answers.Select(answer => answer.Sys.Id), recAnswer => recAnswer.Sys.Id).Any()
Answers.IntersectBy(answers.Select(answer => answer.Sys.Id), chunkAns => chunkAns.Sys.Id)
```

this could become something like
```csharp
public IEnumerable<Answer> GetMatchedAnswers(IEnumerable<Answer> answers, IEnumerable<string> answerIds)
=> answers.IntersectBy(answerIds, answer => answer.Sys.Id);

then you have line 18 becoming

```csharp
GetMatchedAnswers(recommendationSection.Answers, answerIds).Any()
```

and line 23

```csharp
recommendationChunks.Where(chunk => GetMatchedAnswers(chunk.Answers, answerIds).Any())
```

You _could_ maybe also include the `.Any()` in the refactored method as well but without it you have more flexibility.

Reply:
P.S. thinking about it more you could abstract that `GetMatchedAnswers` (which should probably be `GetMatchingAnswers` anyway really) to something like

```csharp
public IEnumerable<TContentfulEntity> GetMatchingEntities(IEnumerable<TContentfulEntity> entities, IEnumerable<string> entityIdsToFind)
where TContentfulEntity : ContentComponent
=> entities.IntersectBy(entityIdsToFind, entity => entity.Sys.Id);
```